### PR TITLE
CVSB-695

### PR DESCRIPTION
**Pull request by vladddragomir**

### CODE REVIEW

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  onEdit() {
+    const actionSheet = this.actionSheetCtrl.create({
+      title: '',
+      buttons: [
+        {
+          text: 'Kilometres',
+          handler: () => {
+            this.odometerMetric = 'Kilometres';
```
Comment (danieliordache) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: you need to capitalise the first letter ? if not, use the lowerCase

File: src/providers/vehicle/vehicle.service.ts [hunk 2]
```diff
@@ -29,4 +31,8 @@ export class VehicleService {
     return lastCertificateExpirationDate;
   }
 
+  formatOdometerReadingValue(string: string) {
+    return string.replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (danieliordache) [inline src/providers/vehicle/vehicle.service.ts]: move it to a separate class. For example:

export class CommonRegExp{
  public static get ODOMETER_VALUE(): RegExp {
    return /\B(?=(\d{3})+(?!\d))/g;
  }
}

then use it as CommonRegExp.ODOMETER_VALUE.

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 1 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/providers/vehicle/vehicle.service.ts (hunk 2 above)
Comment (vladddragomir) [reply src/providers/vehicle/vehicle.service.ts]: done

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  @ViewChild('valueInput') valueInput: TextInput;
+
+  constructor(private actionSheetCtrl: ActionSheetController,
+              private navCtrl: NavController,
+              private navParams: NavParams,
+              private cdRef: ChangeDetectorRef) {
+    this.vehicle = this.navParams.get('vehicle');
+    this.odometerReading = this.vehicle.odometerReading.length ? this.vehicle.odometerReading : '';
+    this.odometerMetric = this.vehicle.odometerMetric.length ? this.vehicle.odometerMetric : 'kilometres';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: can the 'kilometres' be moved to an enum?

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  onEdit() {
+    const actionSheet = this.actionSheetCtrl.create({
+      title: '',
+      buttons: [
+        {
+          text: 'Kilometres',
+          handler: () => {
+            this.odometerMetric = 'kilometres';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: enum?

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          }
+        },
+        {
+          text: 'Miles',
+          handler: () => {
+            this.odometerMetric = 'miles';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: enum?

File: src/pages/testing/test-creation/test-create/test-create.spec.ts [hunk 6]
```diff
@@ -77,4 +77,23 @@ describe('Component: TestCreatePage', () => {
     expect(component.isTestAbandoned(addedVehicleTest)).toBeTruthy();
   });
 
+  it('should say either a test has odometer data or not', () => {
+    component.ngOnInit();
+    component.testReport.vehicles.push(vehicle);
+
+    expect(component.doesOdometerDataExist(0)).toBeFalsy();
+    component.testReport.vehicles[0].odometerReading = '1234';
+    component.testReport.vehicles[0].odometerMetric = 'miles';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/test-create/test-create.spec.ts]: enum?

File: src/pages/testing/test-creation/test-create/test-create.spec.ts [hunk 7]
```diff
@@ -77,4 +77,23 @@ describe('Component: TestCreatePage', () => {
...
+    expect(component.doesOdometerDataExist(0)).toBeTruthy();
+  });
+
+  it('should return correctly formatted string of odometer data', () => {
+    component.ngOnInit();
+    component.testReport.vehicles.push(vehicle);
+    component.testReport.vehicles[0].odometerReading = '1234';
+    component.testReport.vehicles[0].odometerMetric = 'miles';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/test-create/test-create.spec.ts]: enum?

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 8]
```diff
@@ -35,6 +35,19 @@ export class TestCreatePage implements OnInit {
     }
   }
 
+  doesOdometerDataExist(index: number) {
+    return this.testReport.vehicles[index].odometerReading.length && this.testReport.vehicles[index].odometerMetric.length;
+  }
+
+  getOdometerStringToBeDisplayed(index: number) {
+    if (this.doesOdometerDataExist(index)) {
+      let unit = this.testReport.vehicles[index].odometerMetric === 'kilometres' ? 'km' : 'mi';
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/test-create/test-create.ts]: can you use an enum for the 'kilometres'?

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 3 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 5 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 4 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/pages/testing/test-creation/test-create/test-create.spec.ts (hunk 6 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-create/test-create.spec.ts]: done

File: src/pages/testing/test-creation/test-create/test-create.spec.ts (hunk 7 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-create/test-create.spec.ts]: done

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 8 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-create/test-create.ts]: done

**Pull request merged**